### PR TITLE
Use Arcade to manage .NET Core installation and remove custom logger download

### PR DESCRIPTION
**Pull request by natemcmaster**
Removes copies of our dotnet-install scripts in favor of a new Arcade feature that installs .NET Core runtimes for us.

(kudos to @chcosta)

### CODE REVIEW

**Comment by natemcmaster** [comment]
Blocked on https://github.com/dotnet/arcade/issues/2725

File: global.json [hunk 1]
```diff
@@ -3,7 +3,12 @@
     "version": "3.0.100-preview5-011568"
   },
   "tools": {
-    "dotnet": "3.0.100-preview5-011568"
+    "dotnet": "3.0.100-preview5-011568",
+    "runtimes": {
+      "dotnet": [
+        "$(MicrosoftNETCoreAppPackageVersion)"
```
Comment (JunTaoLuo) [inline global.json]: global.json can now resolve version numbers from MSBuild files?

Comment (natemcmaster) [reply global.json]: It's a custom Arcade feature. It's not generally available to the rest of the world.

**Comment by natemcmaster** [comment]
Now blocked on https://github.com/dotnet/arcade/issues/2730

**Comment by natemcmaster** [comment]
CI checks on Windows timeout after an hour, which is bizarre. It looks like the new installer is getting the right bits, but I wonder if it's also doing something that is causing .NET Core to slow down like 100x on Windows.

Don't merge yet - let's investigate what's going on. Might be related to https://github.com/dotnet/arcade/issues/2744

**Pull request merged**